ledger/trie/iterator.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package trie
+
+import (
+	"github.com/onflow/flow-go/ledger"
+)
+
+// NodeIterator is an iterator over the nodes in a trie.
+// It guarantees a DESCENDANTS-FIRST-RELATIONSHIP in the sequence of nodes it generates:
+//   * Consider the sequence of nodes, in the order they are generated by NodeIterator.
+//     Let `node[k]` denote the node with index `k` in this sequence.
+//   * Descendents-First-Relationship means that for any `node[k]`, all its descendents
```

Comment:
Isn't that descendants instead of descendents?

Reply:
```suggestion
//   * Descendants-First-Relationship means that for any `node[k]`, all its descendants
```